command/services/config.go

```diff
@@ -71,6 +71,16 @@ func serviceToAgentService(svc *structs.ServiceDefinition) (*api.AgentServiceReg
 		result.Check = nil
 	}
 
+	// The structs version has non-pointer Proxy.TransparentProxy and Proxy.AccessLogs
+	// The destination has pointers, so we need to set the destination to nil if there
+	// is a zero-value field.
+	if result.Proxy != nil && result.Proxy.TransparentProxy != nil && reflect.DeepEqual(*result.Proxy.TransparentProxy, api.TransparentProxyConfig{}) {
```

Comment:
I don't know if deep-equal is necessary, since there's no pointers in this struct, but I suppose it's better to be safe for future changes.